includes/gateways/stripe/includes/class-give-stripe-gateway.php

```diff
@@ -24,79 +25,89 @@
-		 */
-		public $payment_intent;
-
-		/**
-		 * Stripe Payment Method Object.
-		 *
-		 * @since  2.5.0
-		 * @access public
-		 *
-		 * @var Give_Stripe_Payment_Method
-		 */
-		public $payment_method;
-
-		/**
-		 * Error Messages.
-		 *
-		 * @since  2.7.0
-		 * @access public
-		 *
-		 * @var array $errorMessages List of error messages.
-		 */
-		public $errorMessages = [];
-
-		/**
-		 * Give_Stripe_Gateway constructor.
-		 *
-		 * @since  2.5.0
-		 * @access public
-		 *
-		 * @return bool|void
-		 */
-		public function __construct() {
-
-			// Set API Version.
-			$this->set_api_version();
-
-			// Call Payment Intent Class to utilize.
-			$this->payment_intent = new Give_Stripe_Payment_Intent();
-			$this->payment_method = new Give_Stripe_Payment_Method();
-
-			add_action( "give_gateway_{$this->id}", [ $this, 'process_payment' ] );
-
-		}
+    class Give_Stripe_Gateway
+    {
+
+        /**
+         * Default Gateway ID.
+         *
+         * @since  2.5.0
+         * @access public
+         *
+         * @var string
+         */
+        public $id;
+
+        /**
+         * Set Latest Stripe Version.
+         *
+         * @since  2.5.0
+         * @access public
+         *
+         * @var string
+         */
+        public $api_version = '2019-05-16';
+
+        /**
+         * Payment Intent.
+         *
+         * @since  2.5.0
+         * @access public
+         *
+         * @var \Stripe\PaymentIntent
+         */
+        public $payment_intent;
+
+        /**
+         * Stripe Payment Method Object.
+         *
+         * @since  2.5.0
+         * @access public
+         *
+         * @var Give_Stripe_Payment_Method
+         */
+        public $payment_method;
+
+        /**
+         * Error Messages.
+         *
+         * @since  2.7.0
+         * @access public
+         *
+         * @var array $errorMessages List of error messages.
+         */
+        public $errorMessages = [];
+
+        /**
+         * Give_Stripe_Gateway constructor.
+         *
+         * @return bool|void
+         * @since  2.5.0
+         * @access public
+         *
+         */
+        public function __construct()
+        {
+            // Set API Version.
+            $this->set_api_version();
+
+            // Call Payment Intent Class to utilize.
+            $this->payment_intent = new Give_Stripe_Payment_Intent();
+            $this->payment_method = new Give_Stripe_Payment_Method();
+
+            /** @var PaymentGatewayRegister $paymentGatewayRegister */
+            $paymentGatewayRegister = give(PaymentGatewayRegister::class);
+
+            /**
+             * @unreleased
+             *
+             * Once a gateway has been moved over to our Payment Gateway API
+             * we do not want the legacy processing to run.
+             */
+            if (!$paymentGatewayRegister->hasPaymentGateway($this->id)) {
+                add_action("give_gateway_{$this->id}", [$this, 'process_payment']);
+            }
```

Comment:
@kjohnson I saw you were returning early to prevent legacy processing.  We don't want to get in a situation where both our legacy and gateway api hooks are running so I added this conditional to make our lives easier.  We'll want do this for other gateway migrations as well.

Reply:
@jonwaldstein that works too. I was wanting to restrict my changes to the gateway that I was migrating, but this is where it would eventually end up - that works for me 👍🏻

Reply:
@jonwaldstein I ended up removing the legacy stripe credit card gateway all together, now that I've migrated the legacy form field markup.